### PR TITLE
Refactor internals. Make the operator more lib-oriented

### PR DESCRIPTION
**Pull request by yalosev**
#### Overview

Refactor everything:
- golangci-lint updated to 1.54.2
- matrix test now have kubernetes versions 1.23-1.27
- `gopkg.in/satori/go.uuid.v1` lib changed with `github.com/gofrs/uuid/v5`
- kube-client lib updated from 0.25.4 to 0.26.9
- `HookMetricsListenPort` was removed
- debug server refactored
- unused Interfaces were removed
- package level structured was moved to unexported
- dump of the `main` queue respects output format
- start of the api-http server was moved to the operator Start() function

#### What this PR does / why we need it

Hide unused structs / methods. Make the shell-operator more lib-oriented to reuse in other projects

#### Special notes for your reviewer

```NOTE
`--hook-metrics-listen-port`  was removed. HTTP handler `/metrics/hooks` integrated into base http server, configured via the `--listen-port` flag
```

### CODE REVIEW

File: pkg/shell-operator/debug_server.go [hunk 1]
```diff
@@ -14,14 +14,14 @@ import (
 	"github.com/flant/shell-operator/pkg/task/dump"
 )
 
-func DefaultDebugServer() *debug.Server {
+func newDebugServer() *debug.Server {
```
Comment (diafour) [inline pkg/shell-operator/debug_server.go:17]: debug_server methods are used during bootstrap in addon_operator 

https://github.com/flant/addon-operator/blob/e4887ca4cd1b248ea7ec6d6f326d96cf77ea9a32/pkg/addon-operator/bootstrap.go#L45
https://github.com/flant/addon-operator/blob/e4887ca4cd1b248ea7ec6d6f326d96cf77ea9a32/pkg/addon-operator/bootstrap.go#L73

File: pkg/shell-operator/bootstrap.go [hunk 2]
```diff
@@ -85,7 +84,7 @@ func RequireExistingDirectory(inDir string) (dir string, err error) {
 	return dir, nil
 }
 
-func EnsureTempDirectory(inDir string) (string, error) {
+func ensureTempDirectory(inDir string) (string, error) {
```
Comment (diafour) [inline pkg/shell-operator/bootstrap.go:87]: This one is also use in addon-operator
https://github.com/flant/addon-operator/blob/e4887ca4cd1b248ea7ec6d6f326d96cf77ea9a32/pkg/addon-operator/bootstrap.go#L34

File: pkg/debug/server.go [hunk 3]
```diff
@@ -11,7 +12,7 @@ import (
 	"github.com/go-chi/chi/v5"
 	"github.com/go-chi/chi/v5/middleware"
 	log "github.com/sirupsen/logrus"
-	"sigs.k8s.io/yaml"
+	"gopkg.in/yaml.v3"
```
Comment (yalosev) [inline pkg/debug/server.go:15]: pay attention

Comment (yalosev) [reply pkg/debug/server.go:15]: ah. Seems ok here. We only marshal values here

File: pkg/kube_events_manager/monitor_config.go [hunk 4]
```diff
@@ -81,24 +81,24 @@ func (c *MonitorConfig) WithFieldSelector(fieldSel *FieldSelector) {
...
+//	}
+//
+//	req := FieldSelectorRequirement{
+//		Field:    field,
+//		Operator: op,
+//		Value:    value,
+//	}
+//
+//	c.FieldSelector.MatchExpressions = append(c.FieldSelector.MatchExpressions, req)
+//}
```
Comment (libmonsoon-dev) [inline pkg/kube_events_manager/monitor_config.go:101]: Maybe this should be removed

Comment (yalosev) [reply pkg/kube_events_manager/monitor_config.go]: Done

**Review by libmonsoon-dev — APPROVED** [review]

**Review by miklezzzz — APPROVED** [review]

**Review by Dmitrykob — APPROVED** [review]

**Pull request merged**